lib/src/facts/resolvers/fips_resolver.cc

```diff
@@ -0,0 +1,23 @@
+#include <internal/facts/resolvers/fips_resolver.hpp>
+#include <facter/facts/collection.hpp>
+#include <facter/facts/fact.hpp>
+#include <facter/facts/map_value.hpp>
+#include <facter/facts/scalar_value.hpp>
+
+using namespace std;
+using namespace facter::facts;
+
+namespace facter { namespace facts { namespace resolvers {
+
+    fips_resolver::fips_resolver() :
+        resolver("fips", {fact::in_fips_mode})
+    {
+    }
+
+    void fips_resolver::resolve(collection& facts)
+    {
+        auto data = collect_data(facts);
+        facts.add(fact::in_fips_mode, make_value<boolean_value>(data.is_fips_mode_enabled));
```

Comment:
Follow-up to my above comment, this is where the validation check would go, where the fact would only be added if the data is valid.

Reply:
Also here the way the tests are wired, if we skip adding the fact altogether there seems to be a check that tries to validate the number of facts expected versus actually found and that starts failing. I did not investigate it further in case it is possible to suppress that and have retained the above implementation to always return the fact.